src/prefect/cli/server.py

```diff
@@ -222,6 +224,62 @@ def prestart_check(base_url: str) -> None:
             )
 
 
+def _validate_multi_worker(workers: int, no_services: bool) -> None:
+    """
+    Validates the configuration for running multiple Prefect server workers.
+
+    Multi-worker mode requires specific infrastructure components to ensure proper
+    coordination and data consistency across worker processes:
+
+    - **Database**: PostgreSQL is required (SQLite causes database locking issues)
+    - **Messaging**: Redis is required for event messaging (in-memory messaging
+      doesn't work across processes)
+
+    Args:
+        workers: The number of worker processes to run. Must be >= 1.
+        no_services: Whether to run without background services. Must be True.
+
+    Raises:
+        exit_with_error: If the configuration is invalid.
+    """
+    from prefect.server.utilities.database import get_dialect
+
+    if workers == 1:
+        return
+
+    if workers < 1:
+        exit_with_error("Number of workers must be >= 1")
+
+    if not no_services:
+        raise exit_with_error("Workers can only be run with --no-services")
+
+    try:
+        connection_url = PREFECT_API_DATABASE_CONNECTION_URL.value()
+        dialect = get_dialect(connection_url)
```

Comment:
Let's use the `settings` object here like you do below.
```suggestion
        dialect = get_dialect(settings.server.database.connetion_url)
```